www/src/components/auth/AuthenticateButton.tsx

```diff
@@ -0,0 +1,39 @@
+import { useAuth0 } from '@auth0/auth0-react';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+type BaseButtonProps = {
+    text: string;
+    onClick: () => void;
+};
+
+const BaseButton: React.FC<BaseButtonProps> = (props) => {
+    return <button onClick={props.onClick}>{props.text}</button>;
+};
+
+const LogoutButton = () => {
+    const { logout } = useAuth0();
+    return <BaseButton text="Logout" onClick={() => logout()} />;
+};
+
+const LoginButton = () => {
+    const { loginWithRedirect } = useAuth0();
+    return <BaseButton text="Login" onClick={() => loginWithRedirect()} />;
+};
+
+const AuthenticateButton: React.FC = () => {
```

Comment:
Nit: Maybe a better name for this is `AuthenticationButton` or `AuthButton`. Better yet, I think this should just export a function that runs the correct login/logout logic as needed. Then separate components can leverage the function as needed. This would separate out auth logic and how we render it. Anyways, we probably don't need to worry about this with this prototype if you don't want.

Reply:
The logic is all in `useAuth0` so yes I think once we get to actually integrating it with the UI, we'll make a separate PR to clean this up